### ISSUE TITLE
TypeError: Cannot read property 'id' of undefined

### ISSUE THREAD

**Issue by silverwind** [sdelements/lets-chat#398]
Just had this crash:

```
lets-chat/app/core/presence/connection-collection.js:41
        return value.user.id;
                         ^
TypeError: Cannot read property 'id' of undefined
    at lets-chat/app/core/presence/connection-collection.js:41:26
    at lets-chat/node_modules/lodash/index.js:2269:21
    at lets-chat/node_modules/lodash/index.js:3053:15
    at baseForOwn (lets-chat/node_modules/lodash/index.js:2058:14)
    at lets-chat/node_modules/lodash/index.js:3023:18
    at baseMap (lets-chat/node_modules/lodash/index.js:2268:7)
    at Function.map (lets-chat/node_modules/lodash/index.js:6516:14)
    at ConnectionCollection.getUserIds (lets-chat/app/core/presence/connection-collection.js:40:21)
    at Room.getUserIds (lets-chat/app/core/presence/room.js:33:29)
    at Room.containsUser (lets-chat/app/core/presence/room.js:41:17)
```

**Comment by silverwind** [comment]
Little bit more detail: I've been running on io.js 1.6.4 while this happened. While I don't think it's related, I've installed node 0.10.38 in parallel now for lets-chat. Will let you know if it still crashes.

**Comment by hhaidar** [comment]
@silverwind thanks, let us know what happens. We don't officially support io.js yet, maybe in the near future though :)

**Comment by silverwind** [comment]
Well it has happened on 0.10. too, so I think it's not related to io.js. I've now set up `forever` to reboot on crash, so it's not that much of an issue, but you should probably take a look.

**Comment by sibartlett** [comment]
I'll take a look this week.

**Comment by silverwind** [comment]
I think it might generally be wise to either recommend a process manager like `forever` or `pm2` in the wiki, or alternatively, do a clean app restart on `uncaughtException`. I was quite surprised that this app did just crash like it did.

**Comment by sibartlett** [comment]
@silverwind can you test the latest code on master and see if you get any more crashes?

For the time-being I've added some error checking - to mitigate the server crashing. I still need to try and figure out the root cause for the error.

We'll consider recommending forever - we use it for our Let's Chat instance.

**Comment by silverwind** [comment]
Updated my instance. I'll be watching the log for any more errors.

After two of these crashes, I noticed another interesting bit: Some room names were strangley mangled, for example, a room name like

````
public
````

came up as (repeating at least a few hundret times)

````
public [lots of whitespace] public [lots of whitespace] public [lots of whitespace] ...
````

The name was thousand characters long and it pushed off the edit buttons on the GUI, so I had manually edit the HTML to be able to actually click the "edit room" button to fix that name. I'm not 100% certain it was caused from this crash, but its probable.
